### PR TITLE
[PEER-616] BE: Dispatcher in contracts

### PR DESCRIPTION
**Pull request by IgorAlexeew**
1. Add first argument (address user) to all actions called from UI - updateUser(address user, bytes32 ipfsHash)

2. Add role for dispatcher

3. Create function, that checks whether sender is dispathcer, or sender is user:
- checks that actual sender has dispatcher role
- use a passed address if all good

### CODE REVIEW

File: contracts/PeeranhaCommunity.sol [hunk 1]
```diff
@@ -35,11 +35,11 @@ contract PeeranhaCommunity is IPeeranhaCommunity, Initializable, NativeMetaTrans
      * - Must be a new community.
      * - Sender must be a admin.
      */
-    function createCommunity(bytes32 ipfsHash, CommunityLib.Tag[] memory tags) public  {
-        address msgSender = _msgSender();
-        peeranhaUser.checkHasRole(msgSender, UserLib.ActionRole.Admin, 0);
+    function createCommunity(address user, bytes32 ipfsHash, CommunityLib.Tag[] memory tags) public  {
+        peeranhaUser.dispatcherCheck(user, _msgSender());
```
Comment (sergeyi23) [inline contracts/PeeranhaCommunity.sol]: Should we make this call only if `user !=  _msgSender())` ?

File: contracts/PeeranhaUser.sol [hunk 2]
```diff
@@ -45,6 +47,7 @@ contract PeeranhaUser is IPeeranhaUser, Initializable, NativeMetaTransaction, Ac
         _setRoleAdmin(PROTOCOL_ADMIN_ROLE, DEFAULT_ADMIN_ROLE);
         // TODO: Uncomment this when contract size will be smaller or for redeploy
         // _setRoleAdmin(PROTOCOL_ADMIN_ROLE, BOT_ROLE);
+        // _setRoleAdmin(PROTOCOL_ADMIN_ROLE, DISPATCHER_ROLE);
```
Comment (sergeyi23) [inline contracts/PeeranhaUser.sol]: Still unable to uncomment this?

**Review by sergeyi23 — CHANGES_REQUESTED** [review]
Does it have all changes for develop?

**Review by sergeyi23 — APPROVED** [review]

File: contracts/libraries/PostLib.sol [hunk 3]
```diff
@@ -339,32 +342,59 @@ library PostLib  {
...
         uint256 postId,
         bytes32 ipfsHash,
-        uint8[] memory tags
+        uint8[] memory tags,
+        uint32 communityId, 
+        PostType postType
     ) public {
         PostContainer storage postContainer = getPostContainer(self, postId);
-        self.peeranhaCommunity.checkTags(postContainer.info.communityId, tags);
+        if(userAddr == postContainer.info.author) {
```
Comment (sergeyi23) [inline contracts/libraries/PostLib.sol]: I don't expect these changes after latest changes for edit post merged to develop

Comment (IgorAlexeew) [reply contracts/libraries/PostLib.sol]: **Fixed**

**Pull request merged**